src/main/java/com/github/manolo8/darkbot/core/IDarkBotAPI.java

```diff
@@ -130,6 +135,28 @@ default void setVisible(boolean visible, boolean fullyHideEnabled) {
     void refine(long refineUtilAddress, OreAPI.Ore ore, int amount);
 
     long callMethod(int index, long... arguments);
+    boolean callMethodChecked(boolean checkName, String signature, int index, long... arguments);
+    boolean callMethodAsync(int index, long... arguments);
+    boolean useItem(Item item);
+    boolean isUseItemSupported();
```

Comment:
this method checks also if is able to use item currently

Reply:
Nothing prevents capabilities changing dynamically in hasCapability, darkboat with darkhook would return true/false to direct movement etc depending on if you enabled it or not in the settings. Essentially override the hasCapability method in kekka adapter, and if it's asking for direct item support do extra logic, otherwise, default to super.hasCapability